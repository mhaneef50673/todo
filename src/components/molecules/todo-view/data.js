export const todolist = [
  {
    description: 'item 1',
    status: 'pending',
  },
  {
    description: 'item 2',
    status: 'completed',
  }
];
