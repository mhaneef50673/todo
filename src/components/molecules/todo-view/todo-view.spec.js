import React from 'react';
import { mount } from 'enzyme';
import { mountToJson} from 'enzyme-to-json';
import TodoView from './index';
import { todolist } from './data';

describe('TodoView Component', () => {
  const onTodoItemClickHandler = jest.fn();
  it('should render the component', () => {
    const tree = mount(
      <TodoView list={todolist} onTodoItemClickHandler={onTodoItemClickHandler} />
    )
    expect(mountToJson(tree)).toMatchSnapshot();
  });
});