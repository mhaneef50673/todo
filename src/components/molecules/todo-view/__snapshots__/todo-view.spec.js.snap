// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoView Component should render the component 1`] = `
<TodoView
  list={
    Array [
      Object {
        "description": "item 1",
        "status": "pending",
      },
      Object {
        "description": "item 2",
        "status": "completed",
      },
    ]
  }
  onTodoItemClickHandler={[MockFunction]}
>
  <h1>
     Todo List 
  </h1>
  <div
    className="todo-view"
  >
    <Memo(TodoItem)
      index={0}
      key="0"
      onTodoItemClickHandler={[MockFunction]}
      status="pending"
    >
      <div
        className="todo-item pending"
        onClick={[Function]}
      >
        <span
          className="text"
        />
        <span
          className="text"
        >
          pending
        </span>
      </div>
    </Memo(TodoItem)>
    <Memo(TodoItem)
      index={1}
      key="1"
      onTodoItemClickHandler={[MockFunction]}
      status="completed"
    >
      <div
        className="todo-item completed"
        onClick={[Function]}
      >
        <span
          className="text"
        />
        <span
          className="text"
        >
          completed
        </span>
      </div>
    </Memo(TodoItem)>
  </div>
  <div
    className="add-item"
  >
    <div
      className="center"
    >
      <Button
        action={[Function]}
        buttonText="Add Item"
        clickHandler={[Function]}
        extraClass="primary"
      >
        <button
          className="primary"
          onClick={[Function]}
          type="button"
        >
          Add Item
        </button>
      </Button>
    </div>
  </div>
  <div
    className="modal-container"
  >
    <Modal
      isVisible={false}
      title="Enter item Details"
    />
  </div>
</TodoView>
`;
